lib/ansible/cli/arguments/option_helpers.py

```diff
@@ -250,8 +250,8 @@ def add_connect_options(parser):
                                help='connect as this user (default=%s)' % C.DEFAULT_REMOTE_USER)
     connect_group.add_argument('-c', '--connection', dest='connection', default=C.DEFAULT_TRANSPORT,
                                help="connection type to use (default=%s)" % C.DEFAULT_TRANSPORT)
-    connect_group.add_argument('-T', '--timeout', default=C.DEFAULT_TIMEOUT, type=int, dest='timeout',
-                               help="override the connection timeout in seconds (default=%s)" % C.DEFAULT_TIMEOUT)
+    connect_group.add_argument('-T', '--timeout', default=None, type=int, dest='timeout',
```

Comment:
Won't this now stop `DEFAULT_TIMEOUT` from working if a user has set that option?

Reply:
the setting for default_timeout are the same as the connection plugins declare so while the 'base' config is ignored, the settings will not be.

Reply:
It would allow those plugins to override `default`, `ini`, and `env`, which would just be ignored currently though. Could we just add a changelog that any plugins using the cli timeout may need to update the documentation?

I don't think there's an easy way to fix this without some potential change for 3rd party plugins. I wanted to fix this issue more generally by marking the default CLI values as sentinel https://github.com/ansible/ansible/pull/77729/files (expand the first file), but that would be a change in behavior for plugin options relying on the CLI default (they would need to define the default they were relying on before).

Reply:
most older plugins that did not include the config options are probably using `play_context.timeout`, which would still reflect the defaults

Reply:
fyi, `winrm `and `psrp` don't use the config system, but not the common entries, so they also currently ignore cli/default_timeout/play_context

`local` uses connection timeout for select on fork response (not sure this is correct) and is not configurable otherwise

Reply:
also just realized, `cli` usage in the config is pretty new and i have only seen in it in core plugins, so unless 3rd party plugins had a `cli` entry in config .. this does not affect them at all.

Reply:
play_context initializes it's timeout via:
```  timeout = FieldAttribute(isa='int', default=C.DEFAULT_TIMEOUT)```
then updates via:

```        
      if context.CLIARGS.get('timeout', False):
            self.timeout = int(context.CLIARGS['timeout'])
```

So this ends up with the same value in the end, now directly on init vs via forced cliargs being always set.

Reply:
aded breaking_changes entry to clog to explain for the corner case in which 'ignoring' the configured timeout was expected.